### PR TITLE
Feature/25 improve date time handling

### LINKED ISSUE #25: Cleaner way to do dates
Make a cleaner way to (de)serialize a \DateTime object, currently this will do it to object level but we need a optional string representation. Possibly a UCT string

### PR DESCRIPTION
**Pull request by brettminnie**
Can't remember why exactly we didn't merge this :)

However all tests are passing and it's up to date

closes #25

### CODE REVIEW

File: src/BreakfastSerializer/Property/TypeHandler/IsDateTime.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+namespace BDBStudios\BreakfastSerializer\Property\TypeHandler;
+
+/**
+ * Interface IsDateTime
+ * @package BDBStudios\BreakfastSerializer\Property\TypeHandler
+ */
+interface IsDateTime
+{
+    const ISO_STRING = '/^(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})((-|\+)\d{4})$/';
```
Comment (eddiejaoude) [inline src/BreakfastSerializer/Property/TypeHandler/IsDateTime.php]: I bet that was fun :smile:

File: test/BreakfastSerializer/Fixtures/TypeHandlers/DateTimeType.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace BDBStudios\BreakfastSerializerTest\Fixtures\TypeHandlers;
+use BDBStudios\BreakfastSerializer\Property\TypeHandler\IsDateTime;
+use BDBStudios\BreakfastSerializer\Property\TypeHandler\DateTimeHandler;
+
+class DateTimeType implements IsDateTime
```
Comment (eddiejaoude) [inline test/BreakfastSerializer/Fixtures/TypeHandlers/DateTimeType.php]: Missing docbloc

File: src/BreakfastSerializer/Property/TypeHandler/DateTimeHandler.php [hunk 3]
```diff
@@ -0,0 +1,43 @@
+<?php
+
+namespace BDBStudios\BreakfastSerializer\Property\TypeHandler;
+
+trait DateTimeHandler
```
Comment (eddiejaoude) [inline src/BreakfastSerializer/Property/TypeHandler/DateTimeHandler.php]: Docbloc

Comment (brettminnie) [inline src/BreakfastSerializer/Property/TypeHandler/DateTimeHandler.php]: Yeah, is why it says do not merge, not even ready for review :D

Have to integrate it into the serializer, and make it bypassable via config
then it is review ready

On Tue, Jul 28, 2015 at 1:20 PM, Eddie Abou-Jaoude <notifications@github.com

File: src/BreakfastSerializer/Property/TypeHandler/IsDateTime.php (hunk 1 above)
Comment (fpietka) [inline src/BreakfastSerializer/Property/TypeHandler/IsDateTime.php]: Small question here, you will need all those captured?

File: test/BreakfastSerializer/Property/TypeHandler/DateTimeHandlerTest.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+use BDBStudios\BreakfastSerializer\Property\TypeHandler\IsDateTime;
+use BDBStudios\BreakfastSerializerTest\Fixtures\TypeHandlers\DateTimeType;
+
+class DateTimeHandler extends \PHPUnit_Framework_TestCase
+{
+    /** @var  DateTimeType */
+    protected $instance;
+
+    const ISO_REGEX = '/^(\d{4})-(\d{2})-(\d{2})T(\d{2}):(\d{2}):(\d{2})((-|\+)\d{4})$/';
```
Comment (fpietka) [inline test/BreakfastSerializer/Property/TypeHandler/DateTimeHandlerTest.php]: `/^\d{4}-\d{2}-\d{2}T\d{2}:\d{2}:\d{2}(-|\+)\d{4}$/` should be enough here.
 I think the same goes for the prior.

Comment (fpietka) [inline test/BreakfastSerializer/Property/TypeHandler/DateTimeHandlerTest.php]: How about:
`/^(19[0-9]{2}|[2-9][0-9]{3})-((0(1|3|5|7|8)|10|12)-(0[1-9]|1[0-9]|2[0-9]|3[0-1])|(0(4|6|9)|11)-(0[1-9]|1[0-9]|2[0-9]|30)|(02)-(0[1-9]|1[0-9]|2[0-9]))T(0[0-9]|1[0-9]|2[0-3])(:[0-5][0-9]){2}[-+]((0[0-9]|1[0-3])([03]0|45)|14:00)$/`

Doesn't check leap year though.

Kidding :grin:

Comment (brettminnie) [inline test/BreakfastSerializer/Property/TypeHandler/DateTimeHandlerTest.php]: I appreciate the concatenation,  though the long form is so perverted I may
have to use it

Comment (brettminnie) [inline test/BreakfastSerializer/Property/TypeHandler/DateTimeHandlerTest.php]: Actually (1|2|3|4... ;-)

File: src/BreakfastSerializer/Plugin/PluginInterface.php [hunk 5]
```diff
@@ -34,9 +34,10 @@ public function deactivate();
     public function isActive();
 
     /**
+     * @param mixed $value
      * @return PluginInterface
```
Comment (eddiejaoude) [inline src/BreakfastSerializer/Plugin/PluginInterface.php]: :+1:

**Comment by eddiejaoude** [comment]
Failed CI 

```
[RuntimeException]                                                           
  Could not load package shopware/shopware in http://packagist.org: [Unexpect  
  edValueException] Could not parse version constraint ~v2.0.0-rc1: Invalid v  
  ersion string "~v2.0.0-rc1"    
```

**Comment by brettminnie** [comment]
Yeah I saw that, can't even find reference to that package

Bizarre

Trying to replicate it locally

**Pull request merged**